tket/tests/Circuit/test_Boxes.cpp

```diff
@@ -989,6 +989,178 @@ SCENARIO("Checking equality", "[boxes]") {
   }
 }
 
+SCENARIO("ToffoliBox", "[boxes]") {
+  GIVEN("No permutation.") {
+    std::map<std::vector<bool>, std::vector<bool>> permutation = {};
+    ToffoliBox tb(permutation);
+    std::vector<Command> tb_circuit_commands = tb.to_circuit()->get_commands();
+    REQUIRE(tb_circuit_commands.size() == 0);
+  }
+  GIVEN("Single qubit basis state permutation") {
+    std::map<std::vector<bool>, std::vector<bool>> permutation;
+    permutation[{0}] = {1};
+    permutation[{1}] = {0};
+    ToffoliBox tb(permutation);
+    std::vector<Command> tb_circuit_commands = tb.to_circuit()->get_commands();
+    REQUIRE(tb_circuit_commands.size() == 1);
+    Command com = tb_circuit_commands[0];
+    REQUIRE(com.get_qubits().size() == 1);
+    REQUIRE(com.get_op_ptr()->get_type() == OpType::X);
+  }
+  GIVEN("Two qubit basis state permutation, one two states cycle") {
+    std::map<std::vector<bool>, std::vector<bool>> permutation;
+    permutation[{0, 0}] = {1, 1};
+    permutation[{1, 1}] = {0, 0};
+    ToffoliBox tb(permutation);
+    Circuit comparison_circuit(2);
+    comparison_circuit.add_op<unsigned>(OpType::X, {1});
```

Comment:
I think these tests would be easier to verify if instead of constructing a `comparison_circuit` and calculating the unitary from it, we just supplied tn expected unitary.

Reply:
Agreed, You'll see with the later tests I just compare the right unitary matrix elements are correct -> I'll do the same for these.